File: sprint1apache/www/site3/calculadora.php
<!DOCTYPE html>
<html lang="es">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Calculadora PHP</title>
</head>
<body>
    <h1>Calculadora</h1>

    <!-- Formulario HTML -->
    <form method="POST" action="calculadora.php">
        <label for="numero1">Número 1:</label>
        <input type="number" name="numero1" id="numero1" required>
        <br><br>

        <label for="numero2">Número 2:</label>
        <input type="number" name="numero2" id="numero2" required>
        <br><br>

        <label for="operacion">Operación:</label>
        <select name="operacion" id="operacion" required>
            <option value="suma">Suma</option>
            <option value="resta">Resta</option>
            <option value="multiplicacion">Multiplicación</option>
            <option value="division">División</option>
        </select>
        <br><br>

        <input type="submit" value="Calcular">
    </form>

    <!-- Procesamiento del formulario en PHP -->
    <?php
    if ($_SERVER["REQUEST_METHOD"] == "POST") {
        $numero1 = $_POST['numero1'];
        $numero2 = $_POST['numero2'];
        $operacion = $_POST['operacion'];
        $resultado = 0;

        if (is_numeric($numero1) && is_numeric($numero2)) {
            switch ($operacion) {
                case 'suma':
                    $resultado = $numero1 + $numero2;
                    echo "<h2>Resultado: $numero1 + $numero2 = $resultado</h2>";
                    break;
                case 'resta':
                    $resultado = $numero1 - $numero2;
                    echo "<h2>Resultado: $numero1 - $numero2 = $resultado</h2>";
                    break;
                case 'multiplicacion':
                    $resultado = $numero1 * $numero2;
                    echo "<h2>Resultado: $numero1 * $numero2 = $resultado</h2>";
                    break;
                case 'division':
                    if ($numero2 != 0) {
                        $resultado = $numero1 / $numero2;
                        echo "<h2>Resultado: $numero1 / $numero2 = $resultado</h2>";
                    } else {
                        echo "<h2>Error: No se puede dividir por cero.</h2>";
                    }
                    break;
                default:
                    echo "<h2>Error: Operación no válida.</h2>";
                    break;
            }
        } else {
            echo "<h2>Error: Por favor, ingresa valores numéricos válidos.</h2>";
        }
    }
    ?>
</body>
</html>

File: sprint3python/calculadora.py
from operaciones import suma, resta, multiplicacion, division

def main():
    while True:
        try:
            # Solicitar dos números al usuario
            numero1 = float(input("Introduce el primer número: "))
            numero2 = float(input("Introduce el segundo número: "))
        except ValueError:
            print("Por favor, introduce un número válido.")
            continue

        # Solicitar el tipo de operación
        print("Operaciones disponibles:")
        print("1 - Suma")
        print("2 - Resta")
        print("3 - Multiplicación")
        print("4 - División")
        operacion = input("Selecciona el tipo de operación (1/2/3/4): ")

        # Ejecutar la operación seleccionada
        if operacion == "1":
            resultado = suma(numero1, numero2)
        elif operacion == "2":
            resultado = resta(numero1, numero2)
        elif operacion == "3":
            resultado = multiplicacion(numero1, numero2)
        elif operacion == "4":
            resultado = division(numero1, numero2)
        else:
            print("Operación no válida.")
            continue

        print(f"El resultado es: {resultado}")

        # Preguntar si el usuario quiere hacer otra operación
        continuar = input("¿Quieres hacer otra operación? (s/n): ").lower()
        if continuar != "s":
            print("Gracias por usar la calculadora. ¡Adiós!")
            break

if __name__ == "__main__":
    main()
